week05/DesignLinkedList.cpp

```diff
@@ -0,0 +1,82 @@
+//TC: O(N)
+//MC: O(N)
+class MyLinkedList {
+public:
+    class ListNode{
+    public:
+        int val;
+        ListNode* next;
+        ListNode(int value, ListNode* nxt=NULL){
+            val = value;
+            next = nxt;
+        }
+    };
+    
+    ListNode* head;
+    
+    MyLinkedList() {
+        head = new ListNode(INT_MIN);
+    }
+    
+    int get(int index) {
+        ListNode *cur = head;
+        for (int i = 0; i <= index; i++){
+            cur = cur->next;
+            if (cur == NULL){
```

Comment:
The avg time complexity could be drastically lowered if you have an early termination system. You could have used a member variable in this class indicating the size of the linked list. Thus, you can handle the invalid indices.